apps/recovery-portal/src/main/webapp/recovery.jsp

```diff
@@ -39,7 +39,10 @@
     String username = request.getParameter("username");
     String confirmationKey = request.getParameter("confirmationKey");
     String callback = request.getParameter("callback");
-    String tenantDomain = request.getParameter("tenantDomain").trim();
+    String tenantDomain = StringUtils.EMPTY;
```

Comment:
Why didn't we set this to the super tenant?

Reply:
When the request comes to **recoverUsernamePost** in **RecoverUsernameApiServiceImpl** the tenant domain is added as super tenant if we send it as an empty string from here.